client/src/components/StateBoundaries.jsx

```diff
@@ -0,0 +1,58 @@
+import data from "../state_boundaries.json";
+import { GeoJsonLayer } from "@deck.gl/layers";
+import { getIncidents } from "../utils/api";
+
+const StateBoundaries = async () => {
+  const counts = await makeStateIncidentCounts();
+  const totCount = await getNumIncidents();
+  const structure = await makeStateIncidentCounts();
+  const layer = new GeoJsonLayer({
+    id: "polygon-layer",
+    data,
+    pickable: true,
+    stroked: true,
+    filled: true,
+    wireframe: true,
+    lineWidthMinPixels: 1,
+    getFillColor: (d) => determineColor(d.properties.NAME, counts, totCount),
+    getLineColor: [90, 80, 80],
+    getLineWidth: 1,
+  });
+
+  return layer;
+};
+
+const determineColor = (state, counts, totCount) => {
+  const count = counts[state];
+
+  // colors for ascending percentiles
+  if (count === undefined) return [211, 202, 197];
+
+  if (count / totCount <= 0.2) return [166, 168, 168];
+
+  if (count / totCount <= 0.4) return [120, 133, 137];
+
+  if (count / totCount <= 0.6) return [79, 102, 110];
+  else return [30, 65, 78];
```

Comment:
This looks pretty good, but I think it would make more sense if we didn't reserve the first color for states with no incidents at all. Maybe we could use a ruleset like this so counts of 0 (undefined) and those in the first 20% percentile are grouped together.
```cpp
// pseudo
if (count == undefined || count / total <= 0.2) return color0;
if (count / total <= 0.4) return color1;
if (count / total <= 0.6) return color2;
if (count / total <= 0.8) return color3;
return color4;
```

would appreciate @rxun's thoughts on this too!